### PR TITLE
Release 3.6 gluster fs cherry e2e

### PR DESCRIPTION
**Pull request by mjudeikis**
Gluster Block support for 3.6 playbooks. Cherry from 3.7 and 3.9.
Follows https://github.com/openshift/openshift-ansible/pull/7470

cc: @jarrpa 

Testing is in progress. some of the code is so far away, so had to skip irrelevant parts of the code as it would require a lot of rework on wider playbooks (file firewall management) 
 ```
[mjudeiki@redhat openshift-ansible]$ git oldest-ancestor upstream/release-3.6
  25f5b7168eb5554dca5b91d63a4d21c724f04812
  
git lg1 upstream/release-3.7...25f5b7168eb5554dca5b91d63a4d21c724f04812 roles/openshift_storage_glusterfs/ > /tmp/glusterfs_release3.6
c * 71f6b2707 -    6 weeks ago GlusterFS - Invoke oc binary with the admin.kubeconfig token rather than default token from $HOME/.kube/config — “DanyC97” (origin/test, origin/release-3.7-glusterFS-cherry-e2e, release-3.7-glusterFS-cherry-e2e)
c * ecdcca69c -    3 weeks ago GlusterFS: Add HEKETI_IGNORE_STALE_OPERATIONS to templates — Jose A. Rivera
c * 8e9ba9e1f -    4 weeks ago GlusterFS: Don't copy non-existant topology file — Jose A. Rivera
c * 599c5db1f -    4 weeks ago Merge pull request #7322 from mjudeikis/gluster-block-iscsi — Scott Dodson
c * bfc2c27c5 -    5 weeks ago Merge pull request #7336 from SaravanaStorageNetwork/fix_s3_service_to_s3_pod — OpenShift Merge Robot
c * cc067b9fb -    5 weeks ago Merge pull request #7320 from SaravanaStorageNetwork/fix_s3_image — Scott Dodson
c * c33fd9ad3 -    5 weeks ago Add s3 and block uninstall sections as well. — Saravanakumar Arumugam
c * adbad4275 -    6 weeks ago Update glusterfs-template:  - Add GB_LOGDIR  - failureThreshold as 50 secs — Saravanakumar Arumugam
c * d4dfb79f2 -    6 weeks ago Merge pull request #7237 from jarrpa/glusterfs-kernel-mod-fix — Scott Dodson
c * c29fc1f1c -    7 weeks ago Fix for support multi-cluster heketi's topology — chinacoolhacker
c * 5cec190b7 -    6 weeks ago Merge pull request #7149 from jarrpa/uninstall-regression — OpenShift Merge Robot
c * f7727449e -    6 weeks ago Add target mount for gluster block — Mangirdas
c * b69bc6b85 -    8 weeks ago kernel module loading fix — Mangirdas
c * 1c5959345 -    7 weeks ago Merge pull request #6918 from SaravanaStorageNetwork/uninstall_playbook — OpenShift Merge Robot
c * b3e254547 -    8 weeks ago Backport changes for glusterfs, heketi, s3 and block templates — Saravanakumar Arumugam
c * ecd474ea5 -    8 weeks ago GlusterFS: Fix block StorageClass heketi route — Jose A. Rivera
c * 9cf1047dc -    8 weeks ago GlusterFS: Minor documentation update — Jose A. Rivera
c * b6e450f86 -    8 weeks ago Merge pull request #6969 from mjudeikis/gluster-dynamic-invetory-bug — OpenShift Merge Robot
c * 1f5a74a3a -    9 weeks ago dynamic inventory bug when group exists but its empty — Mangirdas
c * e4c4d1bd1 -   4 months ago Check for openshift attribute before using it during CNS install. — Jiri Mencak
c * 8f7a40c55 -    9 weeks ago Revert "GlusterFS: Remove image option from heketi command" — Michael Hanselmann
c * e778e27a9 -    9 weeks ago cherry-pick to 3.7 gluster block — Mangirdas (origin/release-3.7-gluster-block-cherry)
c * ba6cd6b89 -   3 months ago Port glusterfs optional default (#6182) to 3.7 stable — David Hocky
s *  31d3bb342 -   4 months ago Merge pull request #6318 from openshift-cherrypick-robot/cherry-pick-6073-to-release-3.7 — Scott Dodson
  |\  
s | * 619c7b0ce -   5 months ago Add check for gluterFS DS to stop restarts — Mangirdas
c * |   6e4eb3286 -   4 months ago Merge pull request #6317 from openshift-cherrypick-robot/cherry-pick-5723-to-release-3.7 — Scott Dodson
  |\ \  
  | * | 6edeab281 -   4 months ago storage_glusterfs: fix typo — Luke Meyer
  | * | 303db892b -   5 months ago Add support for external glusterfs as registry backend — Mangirdas
  | |/  
  * | 356659d6f -   4 months ago retry package operations — Luke Meyer
  |/  
  * 35d202921 -   4 months ago GlusterFS: Remove extraneous line from glusterblock template — Jose A. Rivera
  * e48474a8a -   5 months ago GlusterFS: Add configuration for auto creating block-hosting volumes — Jose A. Rivera
  * d4456167a -   6 months ago GlusterFS: Add gluster-s3 functionality — Jose A. Rivera (upstream/CNS36)
  * 7f99212ec -   6 months ago GlusterFS: Add glusterblock functionality — Jose A. Rivera
  * c25a95a3b -   6 months ago GlusterFS: Update heketi templates for latest version — Jose A. Rivera
  * 8675b610a -   6 months ago GlusterFS: Specify resource requests — Jose A. Rivera
  * 54d4ea101 -   5 months ago Merge pull request #5715 from jarrpa/registry-storage-doc-fix — Scott Dodson
  |\  
  | * 881607e61 -   6 months ago GlusterFS: Fix registry storage documentation — Jose A. Rivera
s * |   416282af7 -   5 months ago Merge pull request #5867 from chinacoolhacker/master — OpenShift Merge Robot
  |\ \  
  | * | 58357db1b -   5 months ago Glusterfs storage templates for v1.5 added — chinacoolhacker
  * | |   0b0dea682 -   5 months ago Merge pull request #5769 from jarrpa/image-revert — OpenShift Merge Robot
  |\ \ \  
  | * | | 3d3853836 -   6 months ago GlusterFS: Remove image option from heketi command — Jose A. Rivera
  | |/ /  
  * | |   2d0365bfa -   5 months ago Merge pull request #5720 from sdodson/issue5108 — Scott Dodson
  |\ \ \  
  | |/ /  
  |/| |   
  | * | bbafaeb91 -   6 months ago Add dm_thin_pool for gluster use — Scott Dodson
  | |/  
  * | b8879b5c6 -   6 months ago GlusterFS: remove topology reference from deploy-heketi — Jose A. Rivera
  |/  
  *   d937aa598 -   6 months ago Merge pull request #5562 from ttindell2/master — OpenShift Merge Robot
  |\  
s | * 0ca3358e0 -   6 months ago Add --image flag to setup-openshift-heketi-storage — Tim Tindell
s * | 4ab3e20e3 -   6 months ago GlusterFS: make ServiceAccounts privileged when either glusterfs or heketi is native — Jose A. Rivera
s * |   33ac86141 -   6 months ago Merge pull request #5415 from ttindell2/master — OpenShift Merge Robot
  |\ \  
  | |/  
s | * 62977f65b -   7 months ago GlusterFS can now be run more than once. Ability to add devices to nodes — Tim Tindell
s * |   d07abc3da -   7 months ago Merge pull request #5151 from jarrpa/gluster-ports — Scott Dodson
  |\ \   
  | * | f00c711b1 -   8 months ago GlusterFS: Correct firewall port names — Jose A. Rivera
s * | | d81b5d69e -   7 months ago GlusterFS: Allow option to use or ignore default node selectors — Jose A. Rivera
s * | | 28d9630d5 -   7 months ago GlusterFS: Clarify heketi URL documentation — Jose A. Rivera
s * | | 29a6b62a2 -   7 months ago GlusterFS: Add files/templates for v3.7 — Jose A. Rivera
  | |/  
  |/|   
* | 3a633130f -   7 months ago Fix typo in variable names for glusterfs firewall configuration — Vasily Chekalkin
* | d9d39f333 -   7 months ago Default to global setting for firewall. — Kenny Woodson
* |   affdabe9d -   7 months ago Merge pull request #5082 from jarrpa/glusterfs-readme-note — Scott Dodson
  |\ \  
  | |/  
  |/|   
s | * 68df4abe2 -   8 months ago GlusterFS: add minor README note for #5071 — Jose A. Rivera
s * |   2dd904fee -   8 months ago Merge pull request #5072 from kwoodson/glusterfs_fw — Scott Dodson
s |\ \  
s | * | 6ff354437 -   8 months ago Moving firewall rules under the role to work with refactor. — Kenny Woodson
s | |/  
s * | 94c195bb5 -   9 months ago Removing dependencies for openshift_repos and setting them up early in the cluster build. — Kenny Woodson
s |/  
s *   ef423c067 -   8 months ago Merge pull request #4909 from jarrpa/glusterfs_name-fix — Scott Dodson
s |\  
s | * 5e838f556 -   8 months ago GlusterFS: Default glusterfs_name in loop items. — Jose A. Rivera
s * | c89a8657e -   8 months ago GlusterFS: Don't use /dev/null for empty file. — Jose A. Rivera
s * | 5d78e74da -   8 months ago GlusterFS: Copy SSH private key to master node. — Jose A. Rivera
s * | d46eeae50 -   8 months ago GlusterFS: Check for namespace if deploying a StorageClass — Jose A. Rivera
s * | bf7bd5fb7 -   8 months ago GlusterFS: Fix variable names in defaults. — Jose A. Rivera
s |/  
s * 2e8005d1f -   8 months ago GlusterFS: Use default namespace when not native. — Jose A. Rivera
s * ccbad9c64 -   9 months ago GlusterFS: Create registry storage svc and ep in registry namespace — Jose A. Rivera
s * 9820edba1 -   9 months ago GlusterFS: Pass all booleans through bool filter. — Jose A. Rivera
s * 2a0c64c41 -   9 months ago GlusterFS: Create in custom namespace by default — Jose A. Rivera
s * 40f72f4b0 -   9 months ago GlusterFS: Make heketi-cli command configurable — Jose A. Rivera
s * 11d153a8a -   9 months ago GlusterFS: Reintroduce heketi-cli check for non-native heketi — Jose A. Rivera
s * 0624a1f36 -   9 months ago GlusterFS: Bug fixes for external GlusterFS nodes — Jose A. Rivera
```

### CODE REVIEW

File: roles/openshift_sanitize_inventory/tasks/unsupported.yml [hunk 1]
```diff
@@ -18,3 +18,49 @@
     msg: |-
       The NetworkManager hook is considered a critical part of the DNS
       infrastructure.
+
+- set_fact:
+    __using_dynamic: True
+  when:
+  - hostvars[inventory_hostname][item] in ['dynamic']
+  with_items:
+  - "{{ hostvars[inventory_hostname] | vars_with_pattern(pattern='openshift_.*_storage_kind') }}"
```
Comment (mjudeikis) [inline roles/openshift_sanitize_inventory/tasks/unsupported.yml]: this is failing

File: playbooks/common/openshift-glusterfs/config.yml [hunk 2]
```diff
@@ -26,15 +26,15 @@
   - role: os_firewall
     when:
     - openshift_storage_glusterfs_is_native | default(True) | bool
-  tasks:
+    static: true
   - include_role:
```
Comment (mjudeikis) [inline playbooks/common/openshift-glusterfs/config.yml]: add task:

File: playbooks/common/openshift-glusterfs/config.yml [hunk 3]
```diff
@@ -26,15 +26,15 @@
   - role: os_firewall
     when:
     - openshift_storage_glusterfs_is_native | default(True) | bool
-  tasks:
+    static: true
```
Comment (mjudeikis) [inline playbooks/common/openshift-glusterfs/config.yml]: add tasks

File: roles/openshift_sanitize_inventory/tasks/unsupported.yml (hunk 1 above)
Comment (jarrpa) [reply roles/openshift_sanitize_inventory/tasks/unsupported.yml]: I think we can just omit most if not all the content in this file.

**Comment by jarrpa** [comment]
@mjudeikis Youalready caught most of the things I would have called out. Just one small comment made.

**Comment by mjudeikis** [comment]
@jarrpa, Still testing in my lab. And need to check if any other PR is needed as I might miss something.

File: roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml [hunk 4]
```diff
@@ -313,7 +314,10 @@
   - glusterfs_storageclass or glusterfs_s3_deploy
 
 - include: glusterblock_deploy.yml
-  when: glusterfs_block_deploy
+  when:
+  - glusterfs_block_deploy
+  #TODO: Remove this when multipathd will be available on atomic
+  - not openshift_is_atomic | bool
```
Comment (mjudeikis) [inline roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml]: openshift.common.is_atomic

File: roles/openshift_hosted/tasks/registry/registry.yml [hunk 5]
```diff
@@ -55,6 +55,7 @@
       action: put
     openshift_hosted_registry_force:
```
Comment (mjudeikis) [inline roles/openshift_hosted/tasks/registry/registry.yml]: Add at the top before the block. Endpoints creation requires temp folder
```
- name: Create temp directory for doing work in
  command: mktemp -d /tmp/openshift-hosted-ansible-XXXXXX
  register: mktempHosted
  changed_when: False
  check_mode: no
```

File: playbooks/common/openshift-glusterfs/config.yml [hunk 6]
```diff
@@ -26,15 +26,15 @@
...
+      tasks_from: host_services.yml
+    when: 
     - openshift_storage_glusterfs_is_native | default(True) | bool
     static: true
 
-- name: Open firewall ports for GlusterFS registry nodes
+- name: Configure GlusterFS registry hosts
   hosts: glusterfs_registry
   vars:
     os_firewall_allow:
```
Comment (mjudeikis) [inline playbooks/common/openshift-glusterfs/config.yml]: add firewall: `[root@dhcp46-176 ~]# firewall-cmd --zone=public --add-port=24010/tcp --add-port=3260/tcp --add-port=111/tcp --add-port=22/tcp --add-port=24007/tcp --add-port=24008/tcp --add-port=49152-49664/tcp --permanent`

File: playbooks/common/openshift-glusterfs/config.yml [hunk 7]
```diff
@@ -64,11 +64,7 @@
       name: os_firewall
     when:
     - openshift_storage_glusterfs_registry_is_native | default(True) | bool
-  tasks:
-  - include_role: 
-      name: os_firewall
-    when:
-    - openshift_storage_glusterfs_registry_is_native | default(True) | bool
+    static: true
```
Comment (mjudeikis) [inline playbooks/common/openshift-glusterfs/config.yml]: need second pair of eyes

**Comment by mjudeikis** [comment]
@jarrpa ready for review. for this one, I tried to stick with the minimum.

File: roles/openshift_storage_glusterfs/defaults/main.yml [hunk 8]
```diff
@@ -16,7 +17,7 @@ openshift_storage_glusterfs_block_host_vol_max: 15
 openshift_storage_glusterfs_block_storageclass: False
 openshift_storage_glusterfs_block_storageclass_default: False
 openshift_storage_glusterfs_s3_deploy: True
-openshift_storage_glusterfs_s3_image: "{{ 'rhgs3/rhgs-gluster-s3-server-rhel7' | quote if deployment_type == 'openshift-enterprise' else 'gluster/gluster-object' | quote }}"
+openshift_storage_glusterfs_s3_image: "{{ 'rhgs3/rhgs-s3-server-rhel7' | quote if openshift_deployment_type == 'openshift-enterprise' else 'gluster/gluster-object' | quote }}"
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply roles/openshift_storage_glusterfs/defaults/main.yml]: it does. I think it exist from 3.0 :)

File: playbooks/common/openshift-glusterfs/config.yml [hunk 9]
```diff
@@ -10,7 +10,7 @@
         installer_phase_glusterfs: "In Progress"
       aggregate: false
 
-- name: Open firewall ports for GlusterFS nodes
+- name: Configure GlusterFS hosts
```
Comment (mjudeikis) [inline playbooks/common/openshift-glusterfs/config.yml]: I not sure if this part is right. i mean usage of `include_role` and how its being used in 3.6.

Comment (jarrpa) [reply playbooks/common/openshift-glusterfs/config.yml]: Can you ellaborate on that? Is something failing for you in testing?

Comment (mjudeikis) [reply playbooks/common/openshift-glusterfs/config.yml]: Not a failures. Just I remember we didn't use `import_playbook` before in 3.6, but now I can see a lot of backported changes (not only gluster) using this syntax. And it requires newer ansible version for it. But as we have etcd and master playbooks using those now in 3.6, we should be fine (assumption here).

**Comment by mjudeikis** [comment]
Some tests fail with syntax errors. need to check if its "version incompatibility or genuine issue". TBC

**Comment by mjudeikis** [comment]
Failing on flakes :/ @jarrpa I think this one good to go

**Comment by jarrpa** [comment]
@michaelgugino As @mjudeikis mentioned, the test failures look like flakes. Can we get a pass?

**Comment by mjudeikis** [comment]
@michaelgugino random_reminder_ping

**Pull request merged**